src/solver/application.cpp

```diff
@@ -400,6 +404,11 @@ void Application::readDataForTheStudy(Data::StudyLoadOptions& options)
         throw Error::NoAreas();
     }
 
+    // Prepare time measurement file
+    // study.folderOutput is defined by Study::loadFromFolder, which MUST be called before
+    study.buffer.clear() << study.folderOutput << Yuni::IO::Separator << "time_measurement.txt";
+    mTimeElapsedWriter.initialize(study.buffer, &pTimeElapsedContentHandler);
```

Comment:
Block :
```C++
study.buffer.clear() << study.folderOutput << Yuni::IO::Separator << "time_measurement.txt";
mTimeElapsedWriter.initialize(study.buffer, &pTimeElapsedContentHandler);
```
does not need to be here, but should be placed just before printing file content, in the **Application** destructor.
And that would allow to remove **mTimeElapsedWriter** as an **Application**'s data member, and it also would allow to get rid of the **initialize(...)** in the Writer (and use the writer constructor)

Reply:
However, creating a file in `~Application` is a bit risky since creation may fail. It's better to create that file before the simulation.